scr/main/java/bittercode/model/StoreException.java

```diff
@@ -0,0 +1,65 @@
+package com.bittercode.model;
+
+import java.io.IOException;
+
+import com.bittercode.constant.ResponseCode;
+
+public class StoreException extends IOException {
+
+    private String errorCode;
+    private String errorMessage;
+    private int statusCode;
+
+    public StoreException(String errorMessage) {
+        super(errorMessage);
+        this.errorCode = "BAD_REQUEST";
+        this.setStatusCode(400);
+        this.errorMessage = errorMessage;
+    }
+
+    public StoreException(ResponseCode errorCodes) {
+        super(errorCodes.getMessage());
+        this.statusCode = errorCodes.getCode();
+        this.errorMessage = errorCodes.getMessage();
+        this.setErrorCode(errorCodes.name());
+    }
+
+    public StoreException(String errroCode, String errorMessage) {
+        super(errorMessage);
+        this.errorCode = errroCode;
+        this.errorMessage = errorMessage;
+        this.statusCode = 422;
+    }
+
+    public StoreException(int statusCode, String errorCode, String errorMessage) {
+        super(errorMessage);
+        this.statusCode = statusCode;
+        this.errorCode = errorCode;
+        this.errorMessage = errorMessage;
+    }
+
+    public String getErrorCode() {
+        return null;
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Incorrect return value in getErrorCode method</b></div>
<div id="fix">

The `getErrorCode()` method is returning `null` instead of the `errorCode` field. This might lead to unexpected behavior when trying to retrieve the error code. A similar issue was also found in scr/main/java/bittercode/util/DBUtil.java (line 36).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -41,7 +41,7 @@
  public String getErrorCode() {
 -        return null;
 +        return errorCode;
      }

```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/3#issuecomment-2434572244>#7fa0a4</a></i></small>
</div>


---
🚩 Problem result:
- [ ] This issue was flagged incorrectly